src/main/java/com/dataloom/search/requests/SearchDetails.java

```diff
@@ -0,0 +1,42 @@
+package com.dataloom.search.requests;
+
+import java.io.Serializable;
+import java.util.UUID;
+
+import com.dataloom.client.serialization.SerializationConstants;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class SearchDetails implements Serializable {
+    private static final long serialVersionUID = -6691065251875288358L;
+
+    private final String searchTerm;
+    private final UUID propertyType;
+    private final boolean exactMatch;
+    
+    @JsonCreator
+    public SearchDetails(
+            @JsonProperty( SerializationConstants.SEARCH_TERM ) String searchTerm,
+            @JsonProperty( SerializationConstants.PROPERTY_FIELD ) UUID propertyType,
+            @JsonProperty( SerializationConstants.EXACT ) boolean exactMatch ) {
+        this.searchTerm = searchTerm;
```

Comment:
doesn't the fact that the field is not an optional field imply that it must be present? Otherwise (if this is being passed to an endpoint) it'll throw a 400. I thought we use Google Preconditions to check for specific requirements like if a string or a list isn't allowed to be empty, or things like that.